account_multicurrency_revaluation/wizard/wizard_currency_revaluation.py

```diff
@@ -152,18 +130,16 @@ def _compute_unrealized_currency_gl(self, cr, uid,
 
         @return: updated data for foreign balance plus rate value used
         """
-        context = context or {}
+        context = self._context
 
-        currency_obj = self.pool.get('res.currency')
-        type_id = form.currency_type and form.currency_type.id or False
+        currency_obj = self.env['res.currency']
 
         # Compute unrealized gain loss
         ctx_rate = context.copy()
         ctx_rate['date'] = form.revaluation_date
-        ctx_rate['currency_rate_type_id'] = type_id
-        cp_currency_id = form.journal_id.company_id.currency_id.id
+        cp_currency_id = form.journal_id.company_id.currency_id
 
-        currency = currency_obj.browse(cr, uid, currency_id, context=ctx_rate)
+        currency = currency_obj.with_context(ctx_rate).browse(currency_id)
```

Comment:
In `_get_conversion_rate` in `_compute` (https://github.com/odoo/odoo/blob/8.0/openerp/addons/base/res/res_currency.py#L220-223)
date is applied on both currency. I think you only need to use `with_context` when calling `compute`